src/qiskit_toqm/toqm_latency.py

```diff
@@ -63,37 +64,51 @@ def gen_swap_circuit(s, t):
 
 
 def latencies_from_target(
-    coupling_map,
-    instruction_durations,
+    coupling_map=None,
+    instruction_durations=None,
     basis_gates=None,
     backend_properties=None,
+    target=None,
     normalize_scale=2
 ):
     """
     Generate a list of native ``LatencyDescription`` objects for
     the specified target device.
 
     Args:
-        coupling_map (CouplingMap): CouplingMap of the target backend.
-        instruction_durations (InstructionDurations): Durations for gates
+        coupling_map (Optional[CouplingMap]): CouplingMap of the target backend.
+            Required unless ``target`` is specified.
+        instruction_durations (Optional[InstructionDurations]): Durations for gates
             in the target's basis. Must include durations for all gates
             that appear in input DAGs other than ``swap`` (for which
             durations are calculated through decomposition if not supplied).
+            Required unless ``target`` is specified.
         basis_gates (Optional[List[str]]): The list of basis gates for the
-            target. Must be specified unless ``instruction_durations``
-            contains durations for all swap gates.
+            target. Required unless ``instruction_durations``
+            contains durations for all swap gates or ``target`` is specified.
         backend_properties (Optional[BackendProperties]): The backend
-            properties of the target. Must be specified unless
-            ``instruction_durations`` contains durations for all swap gates.
+            properties of the target. Required unless
+            ``instruction_durations`` contains durations for all swap gates
+            or ``target`` is specified.
+        target (Optional[Target]): The backend transpiler target. If specified,
+            overrides ``coupling_map``, ``instruction_durations``, ``basis_gates``,
+            and ``backend_properties``. All gates that appear in input DAG other
+            than ``swap`` must be supported by the target and have duration
+            information available therein.
         normalize_scale (int): Multiple by this factor when converting
             relative durations to cycle count. The conversion is:
             cycles = ceil(duration * NORMALIZE_SCALE / min_duration)
             where min_duration is the length of the fastest non-zero duration
             instruction on the target.
     """
+    if target is not None:
+        coupling_map = target.build_coupling_map()
+        instruction_durations = target.durations()
+        basis_gates = target.operation_names
```

Comment:
I'm not sure how much it matters for your use case here, but in a target the coupling map and basis gates are not necessarily global. Instructions can be defined on subsets of qubits, so by just doing this you might be ignoring those constraints.

In a normal transpiler pipeline though this will be corrected by basis translation and other passes which are target aware and respect these constraints. Typically routing and layout passes only deal with the global connectivity graph and let other passes adjust for the local constraints. The only reason I'm bringing this up here is because toqm is a little bit different because it's worried about gate durations too.

Reply:
Discussed with @mtreinish offline. Summary is:

`libtoqm` currently assumes that 1Q and 2Q gates are available on all qubits and couplings, respectively. Duration calculation is not an issue, since we calculate timings for each coupling swap by compiling individual single-swap circuits.

To properly handle architectures where different qubits and pairs have different supported instructions, we would need to modify `libtoqm`. While I don't think this would be especially difficult, we'll stick with running a basis translation after routing, which is currently needed for most routing passes.